src/ImageProcessor.Web.Plugins.PostProcessor/PostProcessor.cs

```diff
@@ -41,116 +41,142 @@ internal static class PostProcessor
+                                using (var processResults = await ProcessEx.RunAsync(processStartInfo, cancellationTokenSource.Token).ConfigureAwait(false))
                                 {
-                                    ImageProcessorBootstrapper.Instance.Logger.Log(typeof(PostProcessor), $"Unable to post process image for request {context.Request.Unvalidated.Url}, {processStartInfo.FileName} {processStartInfo.Arguments} exited with error code 1. Original image returned.");
-                                    break;
+                                    if (processResults.ExitCode == 1)
+                                    {
+                                        ImageProcessorBootstrapper.Instance.Logger.Log(typeof(PostProcessor), $"Unable to post process image for request {context.Request.Unvalidated.Url}, {processStartInfo.FileName} {processStartInfo.Arguments} exited with error code 1. Original image returned.");
+                                        break;
+                                    }
+                                }
+
+                                remainingProcesses--;
+
+                                var destinationFileInfo = new FileInfo(destinationFile);
+                                if (destinationFileInfo.Exists)
+                                {
+                                    // Delete the source file
+                                    sourceFileInfo.IsReadOnly = false;
+                                    sourceFileInfo.Delete();
+
+                                    if (remainingProcesses > 0)
+                                    {
+                                        // Use destination file as new source (for the next process)
+                                        destinationFileInfo.MoveTo(sourceFile);
+                                    }
+
+                                    // Swap source for destination
+                                    sourceFileInfo = destinationFileInfo;
+
+                                    // Try to keep the file in memory and ensure it's not indexed
+                                    sourceFileInfo.Attributes |= FileAttributes.Temporary | FileAttributes.NotContentIndexed;
                                 }
                             }
                         }
-                    }
-
-                    // Save result
-                    if (!File.Exists(destinationFile))
-                    {
-                        File.Copy(sourceFile, destinationFile, true);
-                    }
 
-                    var result = new PostProcessingResultEventArgs(destinationFile, length);
-                    if (result.ResultFileSize > 0 && result.Saving > 0)
-                    {
-                        using (var fileStream = File.OpenRead(destinationFile))
+                        // Refresh source file (because it's changed by external processes)
+                        sourceFileInfo.Refresh();
```

Comment:
This should not be required, as the `Exists` and `Length` properties are only accessed below, but this makes debugging a lot easier (because the debugger might already access these properties and return outdated information below).

Reply:
We can wrap this call within `#if DEBUG` and `#endif` statements to save an additional IO call 😄

Reply:
Yeah let's do that. Not that this is the slow part of the process anyway but still, every little helps.

Reply:
Looking at this again, I think it's best to keep the call, as you never know what the external process has done to the source file (it could've deleted it) and because it makes an important decision based on these properties (so it needs to be up-to-date).